controllers/spacebindingcleanup/spacebinding_cleanup_controller_test.go

```diff
@@ -29,200 +31,297 @@ import (
-		require.NoError(t, err)
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara", "redhat", fakeClient).Exists()
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "joe", "redhat", fakeClient).Exists()
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara", "ibm", fakeClient).Exists()
-	})
-
-	t.Run("no SpaceBinding removed when MUR and Space are present", func(t *testing.T) {
-
-		fakeClient := test.NewFakeClient(t, sbLaraRedhatAdmin, sbJoeRedhatView, sbLaraIbmEdit, laraMur, joeMur, ibmSpace, redhatSpace)
-		reconciler := prepareReconciler(t, fakeClient)
-
-		// when
-		res, err := reconciler.Reconcile(context.TODO(), requestFor(sbLaraRedhatAdmin))
-
-		// then
-		require.Equal(t, res.RequeueAfter, time.Duration(0)) // no requeue
-		require.NoError(t, err)
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara", "redhat", fakeClient).Exists()
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "joe", "redhat", fakeClient).Exists()
-		sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara", "ibm", fakeClient).Exists()
-	})
-
-	t.Run("fails while getting the bound resource", func(t *testing.T) {
-
-		for _, boundResourceName := range []string{"lara", "redhat"} {
-			fakeClient := test.NewFakeClient(t, sbLaraRedhatAdmin, redhatSpace, laraMur)
-			reconciler := prepareReconciler(t, fakeClient)
-			fakeClient.MockGet = func(ctx context.Context, key runtimeclient.ObjectKey, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
-				if key.Name == boundResourceName {
-					return fmt.Errorf("some error")
-				}
-				return fakeClient.Client.Get(ctx, key, obj, opts...)
-			}
+func TestDeleteSpaceBindingForKubeSawAuthenticated(t *testing.T) {
+	testCases := []struct {
+		name                  string
+		toolchainConfigOption testconfig.ToolchainConfigOption
+		spaceBindingAssert    func(assertion *spacebinding.Assertion)
+	}{
+		{
+			name:                  "with public-viewer disabled",
+			toolchainConfigOption: testconfig.PublicViewerConfig(false),
+			spaceBindingAssert:    func(assertion *sb.Assertion) { assertion.DoesNotExist() },
+		},
+		{
+			name:                  "with public-viewer enabled",
+			toolchainConfigOption: testconfig.PublicViewerConfig(true),
+			spaceBindingAssert:    func(assertion *sb.Assertion) { assertion.Exists() },
+		},
+	}
 
-			// when
-			_, err := reconciler.Reconcile(context.TODO(), requestFor(sbLaraRedhatAdmin))
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			toolchainconfig := commonconfig.NewToolchainConfigObjWithReset(t, tc.toolchainConfigOption)
 
-			// then
-			require.Error(t, err)
-			sb.AssertThatSpaceBinding(t, test.HostOperatorNs, "lara", "redhat", fakeClient).Exists()
-		}
-	})
+			redhatSpace := spacetest.NewSpace(test.HostOperatorNs, "redhat")
+			sbPublicViewerRedhatView := sb.NewSpaceBinding(toolchainv1alpha1.KubesawAuthenticatedUsername, "redhat", "view", "signupD")
```

Comment:
small optimization - you could move those outside the for loop since they don't rely on any of the tc setup variables. But feel free to leave those here if they improve readability .

Reply:
done 36c456d